src/common/njord.ts

```diff
@@ -447,3 +452,131 @@ export const awardPost = async (
     },
   };
 };
+
+export const awardComment = async (
+  props: AwardInput,
+  ctx: AuthContext,
+): Promise<TransactionCreated> => {
+  const [product, comment, userComment] = await queryReadReplica<
+    [
+      Pick<Product, 'id' | 'type'>,
+      Pick<Comment, 'id' | 'userId' | 'postId' | 'post'>,
+      Pick<UserComment, 'awardTransactionId'> | null,
+    ]
+  >(ctx.con, async ({ queryRunner }) => {
+    return Promise.all([
+      queryRunner.manager.getRepository(Product).findOneOrFail({
+        select: ['id', 'type'],
+        where: {
+          id: props.productId,
+        },
+      }),
+      queryRunner.manager.getRepository(Comment).findOneOrFail({
+        select: ['id', 'userId', 'postId'],
+        where: {
+          id: props.entityId,
+        },
+        relations: {
+          post: true,
+        },
+      }),
+      queryRunner.manager.getRepository(UserComment).findOne({
+        select: ['awardTransactionId'],
+        where: {
+          commentId: props.entityId,
+          userId: ctx.userId,
+        },
+      }),
+    ]);
+  });
+
+  await canAward({ ctx, receiverId: comment.userId });
+
+  if (product.type !== ProductType.Award) {
+    throw new ForbiddenError('Can not award this product');
+  }
+
+  if (userComment?.awardTransactionId) {
+    throw new ConflictError('Comment already awarded');
+  }
+
+  const { transaction, transfer } = await ctx.con.transaction(
+    async (entityManager) => {
+      const { note } = props;
+
+      const transaction = await createTransaction({
+        ctx,
+        entityManager,
+        productId: product.id,
+        receiverId: comment.userId,
+        note,
+      });
+
+      if (!transaction.productId) {
+        throw new Error('Product missing from transaction');
+      }
```

Comment:
In what case does this happen?

Reply:
None currently, but I made `productId` optional so just types being safe, I will look to add NOT NULL constraint when we finish the project depending on the needs